internal/middleware/responsetime.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 Canonical.
+
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/canonical/jimm/v3/internal/servermon"
+)
+
+// MeasureResponseTime tracks response time of requests.
+func MeasureResponseTime(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+		next.ServeHTTP(w, r)
```

Comment:
suggestion:
```
start := time.Now()
defer servermon.ResponseTimeHistogram.WithLabelValues(r.URL.Path, r.Method).Observe(time.Since(start))
next.ServeHTTP(w, r)
```

Reply:
Yeah, I like the `defer` idea, but the value of `time.Since(start)` is evaluated immediately so it wouldn't be the actual time duration of the call. Maybe like this:

```go
start := time.Now()
defer func(){
    servermon.ResponseTimeHistogram.WithLabelValues(r.URL.Path, r.Method).Observe(time.Since(start))
}()
next.ServeHTTP(w, r)
```